src/main/java/uk/gov/hmcts/reform/preapi/actuator/PreApiHealthIndicator.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.preapi.actuator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.preapi.media.MediaKindAccountsClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+public class PreApiHealthIndicator implements HealthIndicator {
+    @Value("${media-service}")
+    protected String mediaService;
+
+    @Value("${azure.ingestStorage.accountName}")
+    protected String ingestStorageAccountName;
+
+    @Value("${azure.finalStorage.accountName}")
+    protected String finalStorageAccountName;
+
+    public final MediaKindAccountsClient mediaKindAccountsClient;
+
+    @Autowired
+    public PreApiHealthIndicator(@Lazy MediaKindAccountsClient mediaKindAccountsClient) {
+        this.mediaKindAccountsClient = mediaKindAccountsClient;
+    }
+
+    @Override
+    public Health health() {
+        if (!mediaService.equals("MediaKind")) {
+            return Health.up().build();
+        }
+        try {
+            var details = checkMediaKindConnections();
+            return Health.up().withDetail("mediakindConnections", details).build();
+        } catch (Exception e) {
+            log.error("Encountered an error when attempting to check MK storage account connections: {}",
+                      e.getMessage());
+        }
+
+        return Health.up().withDetail("mediakindConnections", "unknown error").build();
+    }
+
+    private Map<String, Boolean> checkMediaKindConnections() {
+        var storageAccounts = mediaKindAccountsClient.getStorageAccounts();
+        if (storageAccounts.getItems().isEmpty()) {
+            // maybe don't do this for now (will get this always in prod)
+            log.error("MediaKind does not have any storage account connections");
```

Comment:
Does this need to be removed?

Reply:
I forgot about this. It should be fine to keep in (the code wont run in prod). I'll remove the comment :)